test/test_source_tree_rewriter.rb

```diff
@@ -30,14 +32,23 @@ def apply(actions, **policy)
       rewriter.public_send(action, range, *args)
     end
     if diagnostics.empty?
-      rewriter.process
+      yield rewriter
     else
       diags.call
     end
   rescue ::Parser::ClobberingError => e
     [::Parser::ClobberingError, diags.call]
   end
 
+  # Returns either:
+  #  - String (Normal operation)
+  #  - [diagnostic, ...] (Diagnostics)
+  #  - Parser::ClobberingError
+  #
+  def apply(actions, **policy)
+    build(actions, **policy, &:process)
```

Comment:
nit: could you expand `&:process` please? It's uUnclear without looking at the `build` method who is a receiver of the `.process`

Reply:
You are right.